api/query_engine.go

```diff
@@ -73,15 +95,120 @@ func getOptions(aggSettings mdata.AggSettings, minInterval, tsRange uint32) ([]a
 	return options, aggRef
 }
 
+// getOptionsLikeGraphite is like getOptionsLegacy but returns a sorted (by interval, asc) slice of narchives
+// in which we use
+// the actual interval we would need for raw (which is the LCM of the different raw intervals),
+// potentially making the "raw interval" > first rollup interval, so that
+// raw archive might be ordered after a rollup, and where the archive
+// itself knows which one it is, instead of having to rely on index lookup table.
+func getOptionsLikeGraphite(reqs []models.Req, aggSettings mdata.AggSettings, tsRange uint32) ([]narchive, map[uint32]struct{}) {
```

Comment:
could just use `narchives` instead of `[]narchive` if the type is already defined anyway